educator/src/Dscp/Educator/Web/Educator/Queries.hs

```diff
@@ -283,6 +290,27 @@ educatorAddCertificate cert = do
 
         createCertificate (cfiMeta cert) pdf
 
+getCertificateIssuerInfo
+    :: MonadEducatorWeb ctx m
+    => m Pdf.CertificateIssuerInfo
+getCertificateIssuerInfo = view (lensOf @CertificateIssuerResource) >>= \case
+    KnownIssuerInfo info -> return info
+    FromServiceIssuerInfo baseUrl authToken -> do
+        reqResult <- liftIO $ do
+            manager <- newManager tlsManagerSettings
+            initialRequest <- parseRequest $ showBaseUrl baseUrl
+            let request = initialRequest
+                    { requestHeaders =
+                        [ ("Authorization", authToken)
+                        , ("accept", "application/json")
+                        ]
+                    }
+            response <- httpLbs request manager
+            unless (responseStatus response == status200) $
+                error "Non-200 response code from certificate issuer info request"
```

Comment:
This is a totally plausible scenario, because it's about the behavior of a third party, which could be anything. Such cases should not be treated as `error`, instead, an exception should be made.

Reply:
Fair point, I'll go with it